### ISSUE TITLE
Panic when trying to update with no cargo binary installed

### ISSUE THREAD

**Issue by gyscos** [nabijaczleweli/cargo-update#170]
When no binary has been `cargo install`'ed yet (including `cargo-update`), it panics:
```
thread 'main' panicked at 'index not found', src/ops/mod.rs:785:20
```

This can happen when `cargo-update` (and possibly others) are installed system-wide rather than using cargo itself.

This is using cargo-update ~0.8.0~ 8.0.0 on archlinux from the AUR: https://aur.archlinux.org/packages/cargo-update

At least the `.crates.toml` file exists (but is empty):
```
$ wc ~/.cargo/.crates.toml
0 0 0 /home/gyscos/.cargo/.crates.toml
```

**Comment by nabijaczleweli** [comment]
```
$ git show v0.8.0
tag v0.8.0
Tagger: nabijaczleweli <nabijaczleweli@gmail.com>
Date:   Sat Mar 25 20:40:49 2017 +0100
$ git show v0.8.0:src/ops/mod.rs | wc -l
389
```
huh?

**Comment by gyscos** [comment]
Ha ha nevermind turns out it was for another cargo command, cargo-upgrade. Ugh I have a problem with names...

**Comment by nabijaczleweli** [comment]
no no, this *does* provide `cargo install-update` and `cargo install-update-config`, but 0.8.0 is both four years old and doesn't have a line 785 in the file you posted

**Comment by gyscos** [comment]
Ah yes it was 8.0.0, sorry for the typo.

**Comment by nabijaczleweli** [comment]
ah, that makes more sense (and yeah, each time I bump this I mistype it at *least* twice, too)

**Comment by nabijaczleweli** [comment]
Okay, so you have a `.crates.toml` file (in `$HOME/.cargo`, presumably), but "index not found" probably points to the `["v1"]` index going bad (this code is literally untouched since mid-2017, when it inherited the lack of error handling from its first incarnation in 2017. let's just say it's not written defensively nor well).

Would you mind posting the file here?

**Comment by gyscos** [comment]
I updated the top post; this file is empty on my machine.

**Comment by gyscos** [comment]
Ah running `cargo install --list` does generate this empty file.

**Comment by nabijaczleweli** [comment]
Right. I explicitly check for no file but not for a lack of `v1` section. I'd say that this is a format violation on cargo's part, but :v

Would you mind trying the latest `master` branch (at least c57e0a2d7b0f20e1284c00ed9a1344e795c92a11)?

**Comment by gyscos** [comment]
The latest commit does indeed prevent this panic. Thanks!

**Comment by nabijaczleweli** [comment]
Ace, released in [`v8.1.2`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v8.1.2).
